SQF-VM/SQF_commands.c

```diff
@@ -178,98 +140,90 @@ unsigned char is_equal_to(PVM vm, PVALUE l, PVALUE r)
 		return (l->val.i > 0 ? 1 : 0) == (r->val.i > 0 ? 1 : 0);
-	}
-	else if (l->type == ARRAY_TYPE())
-	{
+	} else if (l->type == ARRAY_TYPE()) {
 		arrl = l->val.ptr;
 		arrr = r->val.ptr;
-		if (arrl->top != arrr->top)
-		{
+		if (arrl->top != arrr->top) {
 			return 0;
 		}
-		for (i = 0; i < arrl->top; i++)
-		{
+		for (i = 0; i < arrl->top; i++) {
 			if (!is_equal_to(vm, arrl->data[i], arrr->data[i]))
 				return 0;
 		}
 		return 1;
-	}
-	else if (l->type == STRING_TYPE())
-	{
-		return !strcmp(((PSTRING)l->val.ptr)->val, ((PSTRING)r->val.ptr)->val);
-	}
-	else if (l->type == NAMESPACE_TYPE())
-	{
+	} else if (l->type == STRING_TYPE()) {
+		return !strcmp(((PSTRING) l->val.ptr)->val, ((PSTRING) r->val.ptr)->val);
+	} else if (l->type == NAMESPACE_TYPE()) {
 		return l->val.ptr == r->val.ptr;
-	}
-	else if (l->type == CODE_TYPE() || l->type == WHILE_TYPE())
-	{
+	} else if (l->type == CODE_TYPE() || l->type == WHILE_TYPE()) {
 		codel = l->val.ptr;
 		coder = r->val.ptr;
 
 		if (codel->stack->top != coder->stack->top)
 			return 0;
-		for (i = 0; i < codel->stack->top; i++)
-		{
+		for (i = 0; i < codel->stack->top; i++) {
 			instl = codel->stack->data[i];
 			instr = coder->stack->data[i];
 			if (instl->type != instr->type)
 				return 0;
-			switch (instl->type)
-			{
-				case INST_VALUE:
-					if (!is_equal_to(vm, get_value(vm, vm->stack, instl), get_value(vm, vm->stack, instr)))
-						return 0;
-					break;
-				case INST_LOAD_VAR:
-				case INST_STORE_VAR:
-				case INST_STORE_VAR_LOCAL:
-					if (str_cmpi(get_var_name(vm, vm->stack, instl), -1, get_var_name(vm, vm->stack, instr), -1))
-						return 0;
-					break;
-				case INST_COMMAND:
-					if (get_command(vm, vm->stack, instl) != get_command(vm, vm->stack, instr))
-						return 0;
-					break;
+			switch (instl->type) {
+			case INST_VALUE:
+				if (!is_equal_to(vm, get_value(vm, vm->stack, instl),
+						get_value(vm, vm->stack, instr)))
+					return 0;
+				break;
+			case INST_LOAD_VAR:
+			case INST_STORE_VAR:
+			case INST_STORE_VAR_LOCAL:
+				if (str_cmpi(get_var_name(vm, vm->stack, instl), -1,
+						get_var_name(vm, vm->stack, instr), -1))
+					return 0;
+				break;
+			case INST_COMMAND:
+				if (get_command(vm, vm->stack, instl)
+						!= get_command(vm, vm->stack, instr))
+					return 0;
+				break;
 			}
 		}
 		return 1;
-	}
-	else if (l->type == SIDE_TYPE())
-	{
+	} else if (l->type == SIDE_TYPE()) {
 		return side_equals(l, r);
-	}
-	else if (l->type == OBJECT_TYPE())
-	{
-		if (((POBJECT)l->val.ptr)->inner == 0 && ((POBJECT)r->val.ptr)->inner == 0)
-		{
+	} else if (l->type == OBJECT_TYPE()) {
+		if (((POBJECT) l->val.ptr)->inner == 0
+				&& ((POBJECT) r->val.ptr)->inner == 0) {
 			return 1;
-		}
-		else
-		{
+		} else {
 			return l->val.ptr == r->val.ptr;
 		}
 	}
 	//else if (l->type == WITH_TYPE())
 	//{
 	//	
 	//}
-	else
-	{
+	else {
 		return 0;
 	}
 }
 
+int find(PVM vm, PARRAY array, PVALUE value) {
```

Comment:
Wrong naming, `PARRAY array` and `PVALUE value` should be qualified with a `const` keyword

proposed name change: `array_indexOf`